### PR TITLE
AsReadOnlySpan extension method additions

### PR DESCRIPTION
**Pull request by jswolf19**
Api Proposal #18420  @jkotas @karelz

### CODE REVIEW

File: src/System.Memory/ref/System.Memory.cs [hunk 1]
```diff
@@ -116,6 +116,9 @@ public static class SpanExtensions
         public static ReadOnlySpan<byte> AsBytes<T>(this ReadOnlySpan<T> source) where T : struct { throw null; }
         
         public static ReadOnlySpan<TTo> NonPortableCast<TFrom, TTo>(this ReadOnlySpan<TFrom> source) where TFrom : struct where TTo : struct { throw null; }
+
+        public static ReadOnlySpan<T> AsReadOnlySpan<T>(this T[] array) { throw null; }
+        public static ReadOnlySpan<T> AsReadOnlySpan<T>(this ArraySegment<T> segment) { throw null; }
```
Comment (shiftylogic) [inline src/System.Memory/ref/System.Memory.cs]: Please rename segment to arraySegment to stay consistent with AsSpan above.

File: src/System.Memory/tests/System.Memory.Tests.csproj [hunk 2]
```diff
@@ -11,6 +11,7 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netfx-Release|AnyCPU'" />
   <ItemGroup>
     <Compile Include="AllocationHelper.cs" />
+    <Compile Include="ReadOnlySpan\AsReadOnlySpan.cs" />
```
Comment (shiftylogic) [inline src/System.Memory/tests/System.Memory.Tests.csproj]: Please move this down into the ReadOnlySpan group below so they stay grouped.

File: src/System.Memory/ref/System.Memory.cs (hunk 1 above)
Comment (ahsonkhan) [inline src/System.Memory/ref/System.Memory.cs]: Also, can you move them up with the AsSpan APIs?

**Comment by jswolf19** [comment]
@shiftylogic @ahsonkhan thanks for the review.

On the moving the APIs in the ref file, would you suggest a newline between AsSpan and AsReadOnlySpan or no? (or is that a little too nitpicky ^^)

**Comment by shiftylogic** [comment]
@dotnet-bot test Innerloop netfx Windows_NT Debug x86 Build and Test
@dotnet-bot test Win x64 tests - Debug - Windows.7.Amd64.Open

**Comment by shiftylogic** [comment]
@dotnet-bot test Win x64 tests - Debug - Windows.7.Amd64.Open

**Comment by shiftylogic** [comment]
@dotnet-bot help

**Comment by ahsonkhan** [comment]
> On the moving the APIs in the ref file, would you suggest a newline between AsSpan and AsReadOnlySpan or no? (or is that a little too nitpicky ^^)

It is good as is :)

Thanks for the contribution. Much appreciated!

**Comment by shiftylogic** [comment]
@dotnet-bot  test this please

**Comment by jswolf19** [comment]
@shiftylogic Should I continue merge commits to master into my branch while there are failing checks, or should I just leave the branch alone unless asked to something else?

**Comment by ahsonkhan** [comment]
> Should I continue merge commits to master into my branch while there are failing checks

You shouldn't need to do this. 

> or should I just leave the branch alone unless asked

Yes. I think something else is broken within the CI and the failures are not related to your change (as far as I can tell).

**Comment by danmosemsft** [comment]
@dotnet-bot test innerloop netfx windows_nt debug x86 build and test (https://github.com/dotnet/corefx/pull/20243)

**Comment by danmosemsft** [comment]
@dotnet-bot test Innerloop Tizen armel Release Cross Build please ("build aborted")
@dotnet-bot test Innerloop Ubuntu14.04 arm Release Cross Build please ("build aborted")

I don't know what build aborted means, let's see if it was some server thing.

**Comment by ahsonkhan** [comment]
> I don't know what build aborted means, let's see if it was some server thing.

OK. The two tests you restarted seem to be stuck.

**Comment by danmosemsft** [comment]
@mmitche

**Comment by mmitche** [comment]
victim of maintenance yesterday.
@dotnet-bot test Innerloop Tizen armel Debug Cross Build
@dotnet-bot test Innerloop Ubuntu16.04 arm Debug Cross Build

**Pull request merged**